File: src/my_project_typescript/easy_problems_typescript/from1to50/projection3dShapes.ts
function projectionArea(grid: number[][]): number {

    let projectionXY: number = 0
    let projectionXZ: number = 0
    let projectionYZ: number = 0

    function getColumnMax(grid: number[][], j:number){
        let temp: Array<number> = [];
        for (let i = 0; i< grid.length; i++){
            temp.push(grid[i][j])
        }
        return Math.max.apply(null, temp)
    }

    function getRowMax(grid: number[][], i:number){
        return Math.max.apply(null,grid[i])

    }


    for (let j = 0; j < grid[0].length; j++){
        projectionYZ += getColumnMax(grid,j)
    }

    for (let i = 0; i < grid.length; i++){
        grid[i].map((currentElement) => {
            if (currentElement !== 0){
                projectionXY += 1
            }
        })
        projectionXZ += getRowMax(grid,i)
    }

    return projectionXY + projectionYZ + projectionXZ
};

console.log(projectionArea( [[1,2],[3,4]]))

File: src/my_project/easy_problems/from101to150/projection_3d_shapes.py
from typing import List 

class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:


        projection_xy = 0
        projection_xz = 0
        projection_yz = 0

        for row in grid:
            projection_xy += sum([1 for x in row if x != 0])
            projection_xz += max(row)

        def max_column(grid, j):
            temp = list()
            for i in range(len(grid)):
                temp.append(grid[i][j])
            return max(temp)

        for j in range(len(grid[0])):
            projection_yz += max_column(grid, j)

        return projection_xy + projection_xz + projection_yz


solution = Solution()
print(solution.projectionArea(
[[1,0],[0,2]]))